bcmp/dfu_core.c

```diff
@@ -527,18 +530,37 @@ static void bm_dfu_event_thread(void *parameters) {
   }
 }
 
+static bool is_link_local_multicast(uint8_t *dst_ip) {
+  bool is_ll_multi = false;
+  if (dst_ip != NULL) {
+    // Link-local multicast address is FF02::1
+    uint8_t *ll_multi_bytes = (uint8_t *)&multicast_ll_addr;
+    is_ll_multi = dst_ip[0] == ll_multi_bytes[0] && // FF
+                  dst_ip[1] == ll_multi_bytes[1] && // 02
+                  dst_ip[15] == ll_multi_bytes[15]; // 01
+  }
+  return is_ll_multi;
+}
+
```

Comment:
This seems like an API that could be useful in other locations. Do you think it is worthwhile to place this in the util.h file and associated source file?

Reply:
It occurred to me, but I didn't do it because I tend to wait until there's more than one usage... 🤔 Sure, yeah, thanks for the nudge. Good call.

Reply:
@matt001k Check out commit 0f37eb3 — I also moved the L2 `is_global_multicast` macro to util.{c,h}.